src/main/java/com/ibm/darpc/DaRPCMemPoolImpl.java

```diff
@@ -0,0 +1,159 @@
+package com.ibm.darpc;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileChannel.MapMode;
+import java.util.LinkedList;
+
+import com.ibm.disni.rdma.RdmaEndpoint;
+import com.ibm.disni.rdma.verbs.IbvMr;
+import com.ibm.disni.rdma.verbs.IbvPd;
+import com.ibm.disni.util.MemoryUtils;
+
+public class DaRPCMemPoolImpl implements DaRPCMemPool {
+	private static final int defaultAllocationSize = 16 * 1024 * 1024; // 16MB
+	private final int allocationSize;
+	private final int alignmentSize;
+	private final String hugePagePath;
+	private String hugePageFile;
+	int offset;
+	ByteBuffer byteBuffer;
+	IbvPd pd;
+	IbvMr mr;
+	int access;
+	LinkedList<ByteBuffer> freeList;
+
+	public DaRPCMemPoolImpl(String hugePagePath, int allocationSize, int alignmentSize) {
+		this.allocationSize = allocationSize;
+		this.alignmentSize = alignmentSize;
+		this.hugePagePath = hugePagePath;
+
+		this.access = IbvMr.IBV_ACCESS_LOCAL_WRITE | IbvMr.IBV_ACCESS_REMOTE_WRITE | IbvMr.IBV_ACCESS_REMOTE_READ;
+	}
+
+	public DaRPCMemPoolImpl(String hugePagePath) {
+		this.allocationSize = defaultAllocationSize;
+		this.alignmentSize = 0;
+		this.hugePagePath = hugePagePath;
+
+		this.access = IbvMr.IBV_ACCESS_LOCAL_WRITE | IbvMr.IBV_ACCESS_REMOTE_WRITE | IbvMr.IBV_ACCESS_REMOTE_READ;
+	}
+
+	// allocate a buffer from hugepages
+	ByteBuffer allocateHugePageBuffer() throws IOException {
+		if (hugePagePath == null) {
+			System.out.println("Hugepage path must be set");
+			throw new IOException("Hugepage path must be set");
+		}
```

Comment:
We should check this in the constructor and e.g. throw an IllegalArgumentException

Reply:
Yes, done.